### PR TITLE
Remove `m.login.token` and `m.login.oauth2` UIA types

### PR DESCRIPTION
**Pull request by richvdh**
AFAIK these have never been implemented. oauth2 lacks enough info to be sensibly implemented, and `m.login.token` just doesn't make sense.

### CODE REVIEW

**Comment by turt2live** [comment]
Surely this needs an MSC regardless of whether or not the thing is implemented in popular servers? It certainly feels extremely wrong to just remove something that's been in the spec for a while without giving people a chance to say "actually, I use this".

**Comment by richvdh** [comment]
OH FINE

#2610, #2611.

File: changelogs/client_server/newsfragments/2609.removal [hunk 1]
```diff
@@ -0,0 +1 @@
+Remove unimplemented `m.login.oauth2` and `m.login.token` user-interactive authentication mechanisms.
```
Comment (turt2live) [inline changelogs/client_server/newsfragments/2609.removal:1]: RST

Comment (turt2live) [inline changelogs/client_server/newsfragments/2609.removal:1]:
```suggestion
Remove unimplemented ``m.login.oauth2`` and ``m.login.token`` user-interactive authentication mechanisms.
```

**Comment by richvdh** [comment]
thanks!

**Pull request merged**